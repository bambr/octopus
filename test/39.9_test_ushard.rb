#!/usr/bin/ruby
# coding: utf-8

$: << File.dirname($0)
require '39_test_ushard'

$one.create_shard 1, :POR, "one"
$one.create_object_space 0, :shard => 1, :index => DEFIDX
$one.insert [1,"one"], :shard => 1
$two.create_shard 1, :POR, "one", "two"
sleep 0.1

# switch master
$two.create_shard 1, :POR, "two", "one"
sleep 0.1

$two.create_shard 1, :POR, "two", "three", "one"
sleep 0.1

$two.create_shard 1, :POR, "two", "three"
sleep 0.1

$two_env.env_eval do
  stop
  puts `./octopus --cat 00000000000000000001.snap`.gsub(/ tm:\d+(\.\d+)? /, ' ')
  puts `./octopus --cat 00000000000000000002.xlog`.gsub(/ tm:\d+(\.\d+)? /, ' ')
  puts `grep META octopus.log`.gsub(/^\d+\.\d+ \d+ /, '')
end

