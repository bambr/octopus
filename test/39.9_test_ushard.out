# one.ping()
:pong

# two.ping()
:pong

# one.meta(shard 1 alter por one)
ok

# one.meta(shard 1 obj_space 0 create tree unique string 0)
ok

# one.insert([1, "one"], {:shard=>1})
1

# two.meta(shard 1 alter por one two)
ok

# two.meta(shard 1 alter por two one)
ok

# two.meta(shard 1 alter por two three one)
ok

# two.meta(shard 1 alter por two three)
ok

lsn:1 scn:-1 t:sys/snap_initial ver:0 count:1 flags:0x00000000
lsn:1 shard:1 scn:4 t:sys/shard_create SHARD_CREATE shard_id:1  POR Box count:1 run_crc:0xa628743f master:one repl:two
lsn:1 shard:1 scn:4 t:snap/usr240 CREATE_OBJECT_SPACE n:0 flags:00000003 cardinalty:0 PK: i:0 min_tuple_cardinality:0 cardinality:1 type:5 unique:1 field0:{index:0 type:7 sort:1}
lsn:1 shard:1 scn:4 t:snap/snap_data n:0 <1:"\x01\x00\x00\x00", "one">
lsn:1 shard:1 scn:4 t:sys/shard_final 
lsn:1 scn:-1 t:sys/snap_final 
lsn:2 shard:1 scn:5 t:sys/shard_alter SHARD_ALTER shard_id:1  POR Box count:1 run_crc:0x3a6501e4 master:two repl:one
lsn:3 shard:1 scn:6 t:sys/shard_alter SHARD_ALTER shard_id:1  POR Box count:1 run_crc:0xa628743f master:two repl:three repl:one
lsn:4 shard:1 scn:7 t:sys/shard_alter SHARD_ALTER shard_id:1  POR Box count:1 run_crc:0x00000000 master:two repl:three
241/load shard {1} I> META mode: PROXY, proxy_addr: 'one/127.0.0.1:33013' => init
241/load shard {1} I> META mode: PARTIAL_PROXY, SCN: 4, type: POR, peer: [one, two], proxy_addr: 'one/127.0.0.1:33013' => hot_standby/one/127.0.0.1:33013/init
242/remote_hot_standby {1} I> META mode: PARTIAL_PROXY, SCN: 4, type: POR, peer: [one, two], proxy_addr: 'one/127.0.0.1:33013' => hot_standby/127.0.0.1:33013/connect
241/load shard {1} I> META mode: PARTIAL_PROXY, SCN: 4, type: POR, peer: [one, two], proxy_addr: 'one/127.0.0.1:33013' => wal_final_row
241/load shard {1} I> META mode: PARTIAL_PROXY, SCN: 4, type: POR, peer: [one, two], proxy_addr: 'one/127.0.0.1:33013' => wal_final_row
242/remote_hot_standby {1} I> META mode: PARTIAL_PROXY, SCN: 4, type: POR, peer: [one, two], proxy_addr: 'one/127.0.0.1:33013' => hot_standby/127.0.0.1:33013/ok
242/remote_hot_standby {1} I> META mode: PARTIAL_PROXY, SCN: 4, type: POR, peer: [one, two], proxy_addr: 'one/127.0.0.1:33013' => wal_final_row
242/remote_hot_standby {1} I> META mode: LOCAL, SCN: 4, type: POR, peer: [two, one] => route update, force notify
242/remote_hot_standby {1} I> META mode: LOCAL, SCN: 4, type: POR, peer: [two, one] => primary
232/box_worker {1} I> META mode: LOCAL, SCN: 6, type: POR, peer: [two, three, one] => route update, force notify
232/box_worker {1} I> META mode: LOCAL, SCN: 7, type: POR, peer: [two, three] => route update, force notify
